### ISSUE TITLE
LONG_FILENAME_WRITE_SUPPORT support

### ISSUE THREAD

**Issue by gryzlov** [OctoPrint/OctoPrint-FirmwareUpdater#315]
Please, add LONG_FILENAME_WRITE_SUPPORT for FB Reborn updating

**Comment by benlye** [comment]
Not 100% sure what you need me to do - just allow long file names in the custom file name option?  I don't know if the underlying MarlinBFT protocol will accept them, and that is outside of my control.

**Comment by gryzlov** [comment]
Yes, can you change regexp to support long file names?

**Comment by benlye** [comment]
Yes, I can do that.  In the meantime, if you set the custom filename option to anything other than `firmware.bin` the setting will be added to OctoPrint's YAML configuration file, where you can then manually change it to anything you like.  You just won't be able to edit it in the UI anymore.

**Comment by gryzlov** [comment]
I removed the code

```
if (self.getProfileSetting("flash_method") == "marlinbft" && self.getProfileSetting("marlinbft_use_custom_filename") && !self.checkMarlinBftCustomFileName()) {
                 alert = gettext("The target filename is invalid. Filename must be 8dot3 format.");
             }
```

and i can confirm that long names (in my case - Robin_nano35.bin) are supported and the firmware is successfully updated.

**Comment by benlye** [comment]
Glad you figured it out, but lets keep this open until I've had a chance to add this option.

**Comment by O1Wcreations** [comment]
how do i go about updating my ender 6, bother the firmware and the cr touch firmware? not sure if im suppose to use a flash method or not and, just flash the file to the serial port?

**Comment by benlye** [comment]
> how do i go about updating my ender 6, bother the firmware and the cr touch firmware? not sure if im suppose to use a flash method or not and, just flash the file to the serial port?

I haven't got a clue - I don't have the board. Your best bet is to find a user group for your printer and ask there.

**Comment by benlye** [comment]
Will be addressed in 1.14.0 (available in the devel branch for testing now).

**Comment by jneilliii** [comment]
Should probably update the UI side of things so it doesn't appear as error when specifying long filename and `Cap:LFN_WRITE:1` in capabilities report. I just tested the latest beta version and the process worked flawlessly for me now. Thanks so much for integrating this change.
